### DISCUSSION TITLE
Does ciborium implement support for major type 2 (byte strings)?

### DISCUSSION THREAD

**Discussion by nlfiedler** [enarx/ciborium#123] [Q&A]
The title says it all. Thanks.

**Comment by rjzak** [comment]
Ciborium supports Strings and raw bytes. I'm not sure what else you might be referring to. There are a few examples in the code, both in terms of inline documentation and unit tests.

https://github.com/enarx/ciborium/blob/ea7d483a7dd0ad751b6a9f1afb239df4c11ee020/ciborium-ll/src/lib.rs#L288-L289

https://github.com/enarx/ciborium/blob/ea7d483a7dd0ad751b6a9f1afb239df4c11ee020/ciborium/src/value/mod.rs#L131-L166

**Comment by nlfiedler** [comment]
In section 3.1 of RFC 8949 there is a description of what I'm looking for, referred to as Major type 2 or "byte string". I would like to have a `Vec<u8>` of length 20 be written out as 21 bytes. Currently, if I serialize a `Vec<u8>` of 20 bytes I get a list of individual bytes and the entire length is ~39 bytes. The minicbor crate has support for this, but only if you use their special encoding/decoding API, which is a bit complex for me since I'm trying to serialize a struct that also has `DateTime<Utc>` fields.
